File: Informatică și tehnologia informației/proba practica/rezolvare_2026/Subiectul_I/a.cpp
#include <iostream>
#include <fstream>
#include <algorithm>

using namespace std;

int main() {
    ifstream fin("minim.in");
    ofstream fout("minim.out");
    
    if (!fin) {
        cerr << "Eroare la deschiderea fisierului minim.in" << endl;
        return 1;
    }
    
    int n;
    if (!(fin >> n)) return 0;
    
    long long val_min;
    if (fin >> val_min) {
        for (int i = 1; i < n; ++i) {
            long long x;
            fin >> x;
            if (x < val_min) {
                val_min = x;
            }
        }
        fout << val_min << "\n";
    }
    
    return 0;
}

File: Informatică și tehnologia informației/proba practica/rezolvare_2026/Subiectul_I/a.pas
program MinimA;

var
  fin, fout: text;
  n, i: integer;
  x, val_min: int64;

begin
  assign(fin, 'minim.in');
  reset(fin);
  assign(fout, 'minim.out');
  rewrite(fout);
  
  if eof(fin) then
  begin
    close(fin);
    close(fout);
    exit;
  end;
  
  read(fin, n);
  if not eof(fin) then
  begin
    read(fin, val_min);
    for i := 2 to n do
    begin
      read(fin, x);
      if x < val_min then
        val_min := x;
    end;
    writeln(fout, val_min);
  end;
  
  close(fin);
  close(fout);
end.
